src/auth/restricted-access.tsx

```diff
@@ -0,0 +1,14 @@
+import React from 'react';
+import { PERMISSIONS } from '../enum'
+
+export const RestrictedAccess: React.FC<{ children: JSX.Element, permissionsPage?: PERMISSIONS[], permissionsUser: PERMISSIONS[] }> = ({
+  permissionsPage, permissionsUser, children
+}) => {
+  const hasPermission: boolean | undefined = permissionsPage?.reduce((acc: boolean, current: PERMISSIONS) => {
+    return acc && Boolean(permissionsUser?.includes(current));
+  }, true);
+  if (hasPermission) {
+    return children;
+  }
+  return null;
```

Comment:
I don't think that it is good to return null directly. This means that we would need to have explicit null check in the preceding component.

Reply:
fixed